openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java

```diff
@@ -118,34 +118,38 @@ public String deployPipeline(IngestionPipeline ingestionPipeline, ServiceEntityI
       response = post(deployUrl, pipelinePayload);
       if (response.statusCode() == 200) {
         ingestionPipeline.setDeployed(true);
-        return response.body();
+        return new Status().withCode(200).withReason(response.body()).withPlatform(this.getPlatform());
```

Comment:
@pmbrull lets create IngestionStatus and not use Response status here
`{
id: pipelineId
status: deployed|failed
message: 'successfully deployed or error reason'
}`

Reply:
The `Status` object we are sending here is not from HTTP Responses, its a new Status class where we pass the request code, any extra info (e.g., any processing error), etc.

In this PR I am getting rid of all generic `Response` returns and either we bring up an object (Ingestion Pipeline, Map with logs,...) or the `Status` telling us how the requested operation went

Reply:
we already have pipelineStatus here https://github.com/open-metadata/OpenMetadata/blob/1b7c5e32337bc313eba34c6c5ed86f0d6506d09a/openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json#L16 let's re-use that

Reply:
but that's different no? That shape is used to store how the pipeline runs evolve: for a given ID, first it is running, then success / fail.

The `Status` here is how we send back the response for the service health and adapted the rest of the responses to just use this single class (kill, delete, deploy,...) which basically gives us an OK/KO of the operation, any specific logs or message and the platform being used (Airflow / Argo)

Reply:
how is that different, isn't UI is using that to show the status of a pipeline. 
When the pipeline deploys it's essentially going to in queue/ running -> success or fail. when we do a bulk deploy we want to see if its in queue/running or failed due to some API issues
all we need is to add a message or reason field of why if its failed to run

Reply:
we don't want to keep adding new "Status" field as existing one can do that exact thing with little modification